src/rc_allocator.c

```diff
@@ -769,50 +769,61 @@ rc_allocator_assert_noleaks(rc_allocator *al)
 void
 rc_allocator_print_stats(rc_allocator *al)
 {
-   int64 divider = GiB / al->cfg->io_cfg->extent_size;
-   platform_default_log(
-      "----------------------------------------------------------------\n");
-   platform_default_log(
-      "| Allocator Stats                                              |\n");
-   platform_default_log(
-      "|--------------------------------------------------------------|\n");
-   uint64 curr_gib = al->stats.curr_allocated / divider;
+   // clang-format off
+   const char *dashes = "-------------------------------------------------------------------";
+   platform_default_log("|%s|\n", dashes);
+   platform_default_log("| Allocator Stats                                                   |\n");
+   platform_default_log("|%s|\n", dashes);
+   // clang-format on
+
+   char   bytes_to_str[SIZE_TO_STR_LEN];
+   size_t nbytes = (al->stats.curr_allocated * al->cfg->io_cfg->extent_size);
+   size_to_fmtstr(bytes_to_str, sizeof(bytes_to_str), "(%s)", nbytes);
+
    platform_default_log(
-      "| Currently Allocated: %12lu extents (%4luGiB)          |\n",
+      "| Currently Allocated: %12lu extents %-14s          |\n",
       al->stats.curr_allocated,
-      curr_gib);
-   uint64 max_gib = al->stats.max_allocated / divider;
+      bytes_to_str);
+
+   nbytes = (al->stats.max_allocated * al->cfg->io_cfg->extent_size);
+   size_to_fmtstr(bytes_to_str, sizeof(bytes_to_str), "(%s)", nbytes);
    platform_default_log(
-      "| Max Allocated:       %12lu extents (%4luGiB)          |\n",
+      "| Max Allocated:       %12lu extents %-14s          |\n",
```

Comment:
Example output from `./build/release/bin/unit/splinter_test test_splinter_print_diags` is as below, for this line and for L784 above:

```
| Currently Allocated:        10495 extents (~1.28 GiB)             |
| Max Allocated:              11193 extents (~1.36 GiB)             |
```

Now you get `~1.28 GiB`, whereas previously you'd get `1 GiB`. New output is slightly more accurate.

Reply:
Ditto